pkg/objectcache/applicationprofilecache/applicationprofilecache.go

```diff
@@ -55,91 +36,252 @@ type ContainerCallStackIndex struct {
 }
 
 type ApplicationProfileCacheImpl struct {
-	cfg                 config.Config
-	containerToSlug     maps.SafeMap[string, string]                      // cache the containerID to slug mapping, this will enable a quick lookup of the application profile
-	slugToAppProfile    maps.SafeMap[string, *v1beta1.ApplicationProfile] // cache the application profile
-	slugToContainers    maps.SafeMap[string, mapset.Set[string]]          // cache the containerIDs that belong to the application profile, this will enable removing from cache AP without pods
-	slugToState         maps.SafeMap[string, applicationProfileState]     // cache the containerID to slug mapping, this will enable a quick lookup of the application profile
-	storageClient       versioned.SpdxV1beta1Interface
-	allProfiles         mapset.Set[string] // cache all the application profiles that are ready. this will enable removing from cache AP without pods that are running on the same node
-	userManagedProfiles maps.SafeMap[string, *v1beta1.ApplicationProfile]
-	containerCallStacks maps.SafeMap[string, *ContainerCallStackIndex] // cache the containerID to call stack search tree mapping
-	containerToName     maps.SafeMap[string, string]                   // cache the containerID to container name mapping
+	cfg                            config.Config
+	workloadIDToProfile            maps.SafeMap[string, *v1beta1.ApplicationProfile]
+	containerIDToInfo              maps.SafeMap[string, *ContainerInfo]
```

Comment:
You need to create a struct that includes v1beta1.ApplicationProfile instead of directly referencing it here. The struct can hold info about the current status of the application profile, like error states, learning period (ready/initializing). It is important for the reasons I have told you. You need to update this status in the periodic updates.